src/dswx_sar/region_growing.py

```diff
@@ -0,0 +1,311 @@
+    for loopind, lines_per_block_loop in enumerate(lines_per_block_list):
+        base_dir = os.path.dirname(output_tif_path)
+        fuzzy_base_name = os.path.splitext(
+            os.path.basename(input_tif_path))[0]
+        lines_per_block_loop = min(data_length,
+                                   lines_per_block_loop)
+        pad_shape = (0, 0)
+        block_params = dswx_sar_util.block_param_generator(
+                lines_per_block_loop,
+                data_shape,
+                pad_shape)
+        # run region-growing for blocks in parallel
+        result = Parallel(n_jobs=-1)(delayed(process_region_growing_block)(
+                                            block_param,
+                                            loopind,
+                                            base_dir,
+                                            fuzzy_base_name,
+                                            input_tif_path,
+                                            initial_seed,
+                                            tolerance,
+                                            maxiter)
+                                     for block_param in block_params)
+
+        for block_param, region_grow_block in result:
+            fuzzy_map_temp = \
+                f'{base_dir}/{fuzzy_base_name}_temp_loop_{loopind + 1}.tif'
+
+            dswx_sar_util.write_raster_block(
+                fuzzy_map_temp,
+                region_grow_block,
+                block_param,
+                geotransform=meta_dict['geotransform'],
+                projection=meta_dict['projection'],
+                DataType='float32')
+
+            # In final loop, write the result to output_tif_path
+            if loopind == num_loop - 1:
+                dswx_sar_util.write_raster_block(
+                    output_tif_path,
+                    region_grow_block,
+                    block_param,
+                    geotransform=meta_dict['geotransform'],
+                    projection=meta_dict['projection'])
+
+
+def run(cfg):
+    '''
+    Run region growing with parameters in cfg dictionary
+    '''
+    logger.info('Starting DSWx-S1 Region-Growing')
+
+    t_all = time.time()
+
+    processing_cfg = cfg.groups.processing
+    outputdir = cfg.groups.product_path_group.scratch_path
+    pol_list = processing_cfg.polarizations
+    pol_str = '_'.join(pol_list)
+
+    # Region growing cfg
+    region_growing_cfg = processing_cfg.region_growing
+    region_growing_seed = region_growing_cfg.seed
+    region_growing_tolerance = region_growing_cfg.tolerance
+    region_growing_line_per_block = region_growing_cfg.line_per_block
+
+    fuzzy_tif_str = os.path.join(outputdir,
+                                 f'fuzzy_image_{pol_str}.tif')
+    water_meta = dswx_sar_util.get_meta_from_tif(fuzzy_tif_str)
+    water_tif_str = os.path.join(outputdir,
+                                 f"region_growing_output_binary_{pol_str}.tif")
+    temp_rg_tif_str = os.path.join(outputdir,
```

Comment:
```suggestion
    water_tif_path = os.path.join(outputdir,
                                 f"region_growing_output_binary_{pol_str}.tif")
    temp_rg_tif_path = os.path.join(outputdir,
```
Same comment as above